src/core/lib/iomgr/sockaddr_posix.h

```diff
@@ -47,6 +47,7 @@ typedef struct in6_addr grpc_in6_addr;
 
 #define GRPC_AF_UNSPEC AF_UNSPEC
 #define GRPC_AF_UNIX AF_UNIX
+#define GRPC_AF_VSOCK AF_VSOCK
```

Comment:
Sorry I didn't notice this on the last review pass. AF_VSOCK may not be defined, right? I believe the rest are guaranteed to exist on all the posix platforms we support. I think the relevant #include should be added to this file, and that this should be guarded by `#ifdef GRPC_HAVE_VSOCK`.

Reply:
do you mean to add below code in sockaddr_posix.h also? if so, I can remove all include from all other cc files like vsock.cc/sock_addr_utils.cc.... 
```
#ifdef GRPC_HAVE_VSOCK
#include <linux/vm_sockets.h>
#endif
```
But it seems not aligned with AF_UNIX, for all other cc files reference AF_UNIX, it needs to include "sys/un.h" before include sockaddr.h. I am not sure whether need to broke this rule? could you confirm this?

Reply:
Good question, it looks inconsistent already. Most of the types used in this file are from either `sys/socket.h` or `arpa/inet.h`, which are included in this file. I _think_ the best thing to do is to conditionally include the vsock headers here, define `GRPC_AF_VSOCK` unconditionally, and wrap all uses of `GRPC_AF_VSOCK` in `#ifdef GRPC_HAVE_VSOCK`. Then all files that need to use AF_VSOCK will only have to wrap the usage in guards, not the includes.